addons/hr_holidays/models/hr_holidays.py

```diff
@@ -0,0 +1,420 @@
+        default='remove')
+    parent_id = fields.Many2one('hr.holidays', string='Parent')
+    linked_request_ids = fields.One2many('hr.holidays', 'parent_id', string='Linked Requests')
+    department_id = fields.Many2one('hr.department',
+        related='employee_id.department_id', string='Department', readonly=True, store=True)
+    category_id = fields.Many2one('hr.employee.category',
+        string="Employee Tag", help='Category of Employee', readonly=True,
+        states={'draft': [('readonly', False)], 'confirm': [('readonly', False)]})
+    holiday_type = fields.Selection([
+        ('employee', 'By Employee'),
+        ('category', 'By Employee Tag')
+    ], string='Allocation Mode', readonly=True, required=True,
+        states={'draft': [('readonly', False)], 'confirm': [('readonly', False)]},
+        help='By Employee: Allocation/Request for individual Employee, By Employee Tag: Allocation/Request for group of employees in category',
+        default='employee')
+    manager_id2 = fields.Many2one('hr.employee',
+        string='Second Approval', readonly=True, copy=False,
+        help='This area is automaticly filled by the user who validate the leave with second level (If Leave type need second validation)')
+    double_validation = fields.Boolean(comodel_name='hr.holidays.status',
+        related='holiday_status_id.double_validation', string='Apply Double Validation')
+    can_reset = fields.Boolean(compute='_compute_can_reset', string="Can reset")
+
+    _sql_constraints = [
+        ('type_value', "CHECK( \
+            (holiday_type='employee' AND employee_id IS NOT NULL) or \
+            (holiday_type='category' AND category_id IS NOT NULL) \
+        )", _("The employee or employee category of this request is missing."
+              " Please make sure that your user login is linked to an employee.")),
+        ('date_check2', "CHECK ( \
+            (type='add') OR (date_from <= date_to) \
+            )", _("The start date must be anterior to the end date.")),
+        ('date_check', "CHECK ( number_of_days_temp >= 0 )",
+            _("The number of days must be greater than 0.")),
+    ]
+
+    @api.depends('number_of_days_temp')
+    def _compute_number_of_days(self):
+        for holiday in self:
+            if holiday.type == 'remove':
+                holiday.number_of_days = -holiday.number_of_days_temp
+            else:
+                holiday.number_of_days = holiday.number_of_days_temp
+
+    def _compute_can_reset(self):
+        """User can reset a leave request if it is its own leave request or if
+        he is an Hr Manager. """
+        if self.env.user.has_group('base.group_hr_manager'):
+            for holiday in self:
+                holiday.can_reset = True
+        else:
+            for holiday in self.filtered(lambda h: h.employee_id and h.employee_id.user_id == self.env.user):
+                holiday.can_reset = True
```

Comment:
May be you can optimize the code like:

for holiday in self:
    if self.env.user.has_group('base.group_hr_manager') or holiday.employee_id.user_id == self.env.user:
    	holiday.can_reset = True